src/Modules/CalcOffence.lua

```diff
@@ -5813,4 +5813,33 @@ function calcs.offence(env, actor, activeSkill)
 	output.CullingDPS = output.CombinedDPS * (bestCull - 1)
 	output.ReservationDPS = output.CombinedDPS * (output.ReservationDpsMultiplier - 1)
 	output.CombinedDPS = output.CombinedDPS * bestCull * output.ReservationDpsMultiplier
+
+	-- Kinetic Fusillade: Calculate effective DPS accounting for projectile firing delay
+	if output.KineticFusilladeMaxEffectiveAttackRate then
+		local maxEffectiveAPS = output.KineticFusilladeMaxEffectiveAttackRate
+		if output.Speed and output.Speed > maxEffectiveAPS then
+			-- Attacking faster than the projectiles can fire wasts some DPS
+			local efficiencyRatio = maxEffectiveAPS / output.Speed
+			output.KineticFusilladeEffectiveDPS = output.TotalDPS * efficiencyRatio
+			output.KineticFusilladeWastedDPS = output.TotalDPS - output.KineticFusilladeEffectiveDPS
+		else
+			-- attack rate is within effective limits
+			output.KineticFusilladeEffectiveDPS = output.TotalDPS
+			output.KineticFusilladeWastedDPS = 0
+		end
+		if breakdown then
```

Comment:
Breakdowns can be done on the gem, not needed to be done here. Have a look at Ball Lightning as an example